hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorContainer.java

```diff
@@ -0,0 +1,456 @@
+        }
+
+        tasks.remove(taskName);
+    }
+
+    public void stash(TaskDefinition definition) {
+        if (logger.isFineEnabled()) {
+            logger.finest("[Backup Scheduler: " + name + "][Partition: " + partitionId + "] Stashing " + definition);
+        }
+
+        if (!backups.containsKey(definition.getName())) {
+            BackupTaskDescriptor descriptor = new BackupTaskDescriptor(definition);
+            backups.put(definition.getName(), descriptor);
+        }
+
+        if (logger.isFineEnabled()) {
+            logger.finest("[Scheduler: " + name + "][Partition: " + partitionId + "] Stash size: " + backups.size());
+        }
+    }
+
+    public void unstash(String taskName) {
+        backups.remove(taskName);
+    }
+
+    public Collection<ScheduledTaskDescriptor> getTasks() {
+        return tasks.values();
+    }
+
+    public void syncState(String taskName, Map newState) {
+        backups.get(taskName).setMasterState(newState);
+    }
+
+    public boolean shouldParkGetResult(String taskName) {
+        if (partitionId == MEMBER_PARTITION) {
+            // For member owned tasks there is a race condition, so we avoid purposefully parking.
+            // TODO tkountis - Look into the invocation subsystem, to identify root cause.
+            return false;
+        }
+
+        return (tasks.containsKey(taskName) && !tasks.get(taskName).getScheduledFuture().isDone())
+                || backups.containsKey(taskName);
+    }
+
+    public int getDurability() {
+        return durability;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    void promoteStash() {
+        Iterator<Map.Entry<String, BackupTaskDescriptor>> iterator =
+                backups.entrySet().iterator();
+
+        while (iterator.hasNext()) {
+            Map.Entry<String, BackupTaskDescriptor> entry = iterator.next();
+            BackupTaskDescriptor descriptor = entry.getValue();
+            doSchedule(descriptor.getDefinition(), descriptor.getMasterState(), descriptor.getMasterStats());
+            iterator.remove();
+        }
+
+    }
+
+    Map<String, BackupTaskDescriptor> prepareForReplication() {
+        Map<String, BackupTaskDescriptor> replicas = new HashMap<String, BackupTaskDescriptor>();
+
+        Iterator<Map.Entry<String, ScheduledTaskDescriptor>> iterator =
+                tasks.entrySet().iterator();
+
+        while (iterator.hasNext()) {
+            Map.Entry<String, ScheduledTaskDescriptor> entry = iterator.next();
+            ScheduledTaskDescriptor descriptor = entry.getValue();
+
+            try {
+                BackupTaskDescriptor replica = new BackupTaskDescriptor(descriptor.getDefinition());
+                replica.setMasterState(descriptor.getState().get());
+                replica.setMasterStats(descriptor.getStats());
+                replicas.put(entry.getKey(), replica);
+            } finally {
+                descriptor.getLock().set(false);
```

Comment:
This was a left-over. It is now changed to a different approach as discussed with @metanet